kyaml/fn/framework/example2/doc.go

```diff
@@ -1,3 +1,8 @@
+// Package example2 contains tests that illustrate how to use framework.TemplateProcessor
+// to build functions that add resources from ResourceTemplates, or modify resources
+// with ResourcePatchTemplates or ContainerPatchTemplates.
+// TODO: why is this an example package rather than a regular test file?
```

Comment:
Agreed. Why?

Reply:
I still haven't gotten through much of the test audit, but at this point my intent is to move it either into regular tests or regular examples rather than a package.